core/http-auth-event-stream/src/main/java/software/amazon/awssdk/http/auth/eventstream/internal/SigV4DataFramePublisher.java

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.auth.eventstream.internal;
+
+import static software.amazon.awssdk.http.auth.internal.util.HttpChecksumUtils.hash;
+import static software.amazon.awssdk.http.auth.internal.util.SignerConstant.AWS4_CHUNK_SIGNING_ALGORITHM;
+import static software.amazon.awssdk.http.auth.internal.util.SignerUtils.computeSignature;
+import static software.amazon.awssdk.http.auth.internal.util.SignerUtils.deriveSigningKey;
```

Comment:
These all look like internal APIs to `http-auth`, so they can't be used here.

Reply:
Hmmm... that could be troublesome. Perhaps, we should move all of the SigV4 signers into http-auth-aws.